public/service/message.go

```diff
@@ -366,6 +367,34 @@ func (m *Message) BloblangQuery(blobl *bloblang.Executor) (*Message, error) {
 	return nil, nil
 }
 
+// BloblangQueryValue executes a parsed Bloblang mapping on a message and
+// returns the result or an error if the mapping fails.
+func (m *Message) BloblangQueryValue(blobl *bloblang.Executor) (any, error) {
+	uw := blobl.XUnwrapper().(interface {
+		Unwrap() *mapping.Executor
+	}).Unwrap()
+
+	msg := message.Batch{m.part}
+
+	res, err := uw.Exec(query.FunctionContext{
+		Maps:     uw.Maps(),
+		Vars:     map[string]any{},
+		Index:    0,
+		MsgBatch: msg,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	switch res.(type) {
+	case value.Delete:
+		return nil, bloblang.ErrRootDeleted
+	case value.Nothing:
+		return nil, nil
+	}
```

Comment:
I'm not sure what's the most user-friendly behaviour here, but I imagine in some cases they might want to be abel to detect that someone added `root = deleted()` in the config. I did the same for `MessageBatch. BloblangQueryValue()`.